src/pgrn-custom-scan.c

```diff
@@ -593,6 +643,54 @@ PGrnSearchBuildCustomScanConditions(CustomScanState *customScanState,
 	}
 }
 
+static void
+PGrnCustomScanSort(CustomScanState *customScanState)
+{
+	const char *tag = "pgroonga: [custom-scan][sort]";
+	PGrnScanState *state = (PGrnScanState *) customScanState;
+	Relation table = customScanState->ss.ss_currentRelation;
+	int nPathKeys = list_length(state->pathKeys);
+	grn_table_sort_key *sortKeys =
+		(grn_table_sort_key *) palloc(sizeof(grn_table_sort_key) * nPathKeys);
+	ListCell *cell;
+	unsigned int nSortKeys = 0;
+	foreach (cell, state->pathKeys)
+	{
+		PathKey *pathKey = (PathKey *) lfirst(cell);
+		EquivalenceMember *member = linitial(pathKey->pk_eclass->ec_members);
+		Expr *expr = (Expr *) (member->em_expr);
+		if (IsA(expr, Var))
+		{
+			// Support only simple sorting by columns.
+			Var *var = (Var *) expr;
+			const char *name = PGrnTableColumnName(table, var);
+			sortKeys[nSortKeys].key =
+				grn_obj_column(ctx, state->searched, name, strlen(name));
+			if (pathKey->pk_cmptype == COMPARE_LT)
+				sortKeys[nSortKeys].flags = GRN_TABLE_SORT_ASC;
+			else if (pathKey->pk_cmptype == COMPARE_GT)
+				sortKeys[nSortKeys].flags = GRN_TABLE_SORT_DESC;
+			nSortKeys++;
+		}
+		else
+		{
+			char *pathKeyStr = nodeToString(pathKey);
+			elog(DEBUG1, "%s It is an invalid pathkey <%s>", tag, pathKeyStr);
+			pfree(pathKeyStr);
+			continue;
+		}
+	}
+
+	state->sorted = grn_table_create(
+		ctx, NULL, 0, NULL, GRN_OBJ_TABLE_NO_KEY, NULL, state->searched);
+	grn_table_sort(
+		ctx, state->searched, 0, -1, state->sorted, sortKeys, nSortKeys);
+
+	for (unsigned int i = 0; i < nSortKeys; i++)
+		grn_obj_close(ctx, sortKeys[i].key);
```

Comment:
We can use `grn_obj_close()` here because all `.key`s are accessor (`state->searched` doesn't have any column) but, in general, we should use `grn_obj_unlink()` for `grn_obj_column()`. `grn_obj_unlink()` closes an accessor but doesn't close a column. 

```suggestion
		grn_obj_unlink(ctx, sortKeys[i].key);
```